### PR TITLE
Lpal 421 library for api

### PR DESCRIPTION
**Pull request by nickdavis2001**
## Purpose

_Initial work to produce flaskapi library. Please see the README for further info._

## Approach

_Python code, with tests, to create Flask api, configured with optional DB connection, AWS xray config, healthcheck which tests DB connection if applicable. github action to install and test this._ 

## Learning

_Learned more about Flask, and Python's packaging system. Interesting to learn that sqlite is actually built-in to Python and that an in-memory DB test fixture can be constructed._

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

File: flaskapi/demo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# context should be set to dir above demo dir
+FROM python:3.9-alpine
+
+# security fix for expat
+RUN apk upgrade expat
+
+RUN apk add build-base curl libpq-dev
```
Comment (townxelliot) [inline flaskapi/demo/Dockerfile]: Can some or all of these packages be removed after the build is complete? Though it doesn't matter that much for a demo I suppose. But might be good to do it anyway to provide a reference Dockerfile for package users.

File: flaskapi/setup.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from setuptools import setup, find_packages
+
+setup(name='opgflaskapi',
```
Comment (townxelliot) [inline flaskapi/setup.py]: I suggest we put some linting on this. Meanwhile, I would format this with each kwarg on its own line, indented 4 spaces:
```
setup(
    name='opgflaskapi',
    version='1.0', 
    description='base flask api',
    install_requires = [
        'aws_xray_sdk',
        'flask-sqlalchemy',
        'flask',
        'psycopg2-binary',
        'gunicorn',
    ],
    packages=[
        'opgflaskapi',
    ]
)
```

File: flaskapi/setup.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from setuptools import setup, find_packages
+
+setup(name='opgflaskapi', 
+        version='1.0', 
+        description='base flask api',
+        install_requires = ['aws_xray_sdk', 'flask-sqlalchemy', 'flask', 'psycopg2-binary', 'gunicorn'],
```
Comment (townxelliot) [inline flaskapi/setup.py]: Suggest you include version specifiers for each dependency, so we pin to at least a major version of the package, e.g.
```
install_requires = [
    'mydependency>=2,<3'
]
```

**Review by townxelliot — CHANGES_REQUESTED** [review]

File: flaskapi/demo/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# context should be set to dir above demo dir
+FROM python:3.9-alpine
```
Comment (townxelliot) [inline flaskapi/demo/Dockerfile]: I wonder whether Dockerfiles should explicitly define the environment variables they need to work properly? In this case, we need four POSTGRES_* env variables, but that's not clear from the Dockerfile.

File: flaskapi/setup.py (hunk 2 above)
Comment (nickdavis2001) [reply flaskapi/setup.py]: Fixed by running black.  We could put black in pre-commit hook on future PR

File: flaskapi/demo/Dockerfile (hunk 1 above)
Comment (nickdavis2001) [reply flaskapi/demo/Dockerfile]: We do not need curl anymore anymore.  We need libpq, but can safely delete build-base after the build

Comment (townxelliot) [reply flaskapi/demo/Dockerfile]: You probably only need libpq rather than libpq-dev I'm guessing.

File: flaskapi/setup.py (hunk 2 above)
Comment (townxelliot) [reply flaskapi/setup.py]: Can you please raise an issue for that @nickdavis2001? Thanks.

File: flaskapi/demo/Dockerfile (hunk 1 above)
Comment (nickdavis2001) [reply flaskapi/demo/Dockerfile]: This is now outdated as demo is no longer included here

File: flaskapi/demo/Dockerfile (hunk 4 above)
Comment (nickdavis2001) [reply flaskapi/demo/Dockerfile]: now outdated as demo no longer here

File: flaskapi/setup.py (hunk 3 above)
Comment (nickdavis2001) [reply flaskapi/setup.py]: Done

**Review by townxelliot — APPROVED** [review]

**Pull request merged**